Khan/wonder-blocks
# Migrate ModalDialog to Storybook

## Summary:
Using Styleguidist as a foundation, migrate the documentation for
ModalDialog to Storybook.

Styleguidist does not have specific examples for each building
block under modal, but I decided to start adding them all with their own
examples (as opposed to using subcomponents), particularly so that
users can experiment with the control panel. This one was a little
tricky since ModalDialog does not have a visual element.

Storybook publish: 
https://5e1bf4b385e3fb0020b7073c-sbvqecqsiq.chromatic.com/?path=/docs/modal-building-blocks-modaldialog--default

Issue: https://khanacademy.atlassian.net/browse/WB-1141

## Test plan:
- Check that all the documentation makes sense.
- Check there are no typos.
- Interact with the modals to confirm they all behave as expected.
